### PR TITLE
remove unused `$index`

### PR DESCRIPTION
**Pull request by ReenExe**

### CODE REVIEW

File: lib/Elastica/Multi/MultiBuilder.php [hunk 1]
```diff
@@ -38,16 +38,15 @@ private function buildResultSet(Response $childResponse, BaseSearch $search)
      */
     private function buildResultSets(Response $response, $searches)
     {
-        $resultSets = array();
-
         $data = $response->getData();
-        if (!isset($data['responses']) || !is_array($data['responses'])) {
-            return $resultSets;
+        if (empty($data['responses']) || !is_array($data['responses'])) {
```
Comment (ruflin) [inline lib/Elastica/Multi/MultiBuilder.php]: I don't think these two are here the same. If `$data['response] == false` the first one would not catch it.

Overall as we look for an `array` to in the end it seems to be the same. Do we have unit tests here to make sure that is the case?

Comment (ReenExe) [inline lib/Elastica/Multi/MultiBuilder.php]: 1. empty([]) // true
2. empty(false) // true
3. $data = [];
empty($data['responses']) // true

Your case:
$data['responses'] = false;
empty($data['responses']) // true

Comment (ruflin) [inline lib/Elastica/Multi/MultiBuilder.php]: Not sure I get this as:
```
$data['responses'] = false;
empty($data['responses']) // true
!isset($data['responses']) // false
```

Even if his currently the same affect, the question is what our intention is: Do we want to check if the array entry is set or checking if it is empty? These are conceptually two different things and I think it is the first one we are looking for. So I would not change this line.

Comment (ReenExe) [inline lib/Elastica/Multi/MultiBuilder.php]: If see all method - and use `!isset($data['responses'])` instead `empty($data['responses']))` - so we have `foreach ([] as $responseData) {}` so use `empty($data['responses']))` - remove this foreach.
The result of method - same, but in my - without `reset` and `empty foreach`

Comment (ruflin) [inline lib/Elastica/Multi/MultiBuilder.php]: Not sure I can follow your comment above?

Comment (ReenExe) [inline lib/Elastica/Multi/MultiBuilder.php]: In my vesrion - add `early return` that remove foreach on empty array

Comment (ruflin) [inline lib/Elastica/Multi/MultiBuilder.php]: I think the return already existed before?

Comment (ReenExe) [inline lib/Elastica/Multi/MultiBuilder.php]: Yes

Comment (ruflin) [inline lib/Elastica/Multi/MultiBuilder.php]: Not 100% I understand what you mean but best is to just push a new version of what you think should be done which makes it easier to discuss.

Comment (ReenExe) [inline lib/Elastica/Multi/MultiBuilder.php]: Ok, soon re-open

**Comment by ReenExe** [comment]
#1147

**Pull request closed without merging**